cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/SparkRuntimeService.java

```diff
@@ -655,6 +655,16 @@ private Map<String, String> createSubmitConfigs(File localDir, @Nullable String
     prependConfig(configs, "spark.driver.extraJavaOptions", cConf.get(Constants.AppFabric.PROGRAM_JVM_OPTS), " ");
     prependConfig(configs, "spark.executor.extraJavaOptions", cConf.get(Constants.AppFabric.PROGRAM_JVM_OPTS), " ");
 
+    // In Spark v3.2.0 and later, max netty direct memory must be >= spark.network.maxRemoteBlockSizeFetchToMem
+    // See https://github.com/apache/spark/commit/00b63c8dc2170cfbc3f3ea7882dfd417d7fde744#.
+    // So we set max netty direct memory = spark.network.maxRemoteBlockSizeFetchToMem
+    // maxRemoteBlockSizeFetchToMem config is set in ETLSpark.java, DataStreamsSparkLauncher.java
```

Comment:
would be good to reference the jira in case people want more context/info

It also seems like we should set it in the same place that spark.network.maxRemoteBlockSizeFetchToMem is being set. Otherwise if we ever change that value for some reason, this may break again.

Reply:
added reference to the jira. moved the code that sets io.netty.maxDirectMemory into the app where we set spark.network.maxRemoteBlockSizeFetchToMem